src/diem/offchain/funds_pull_pre_approval_command.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import dataclasses
+import typing
+import uuid
+
+from . import FundPullPreApprovalObject, CommandType, CommandRequestObject
+from .command import Command
+from .payment_state import Action
+from .types import new_funds_pull_pre_approval_request
+
+
+@dataclasses.dataclass(frozen=True)
+class FundsPullPreApprovalCommand(Command):
+    my_actor_address: str
+    funds_pull_pre_approval: FundPullPreApprovalObject
+    cid: str = dataclasses.field(default_factory=lambda: str(uuid.uuid4()))
+
+    def command_type(self) -> str:
+        return CommandType.FundPullPreApprovalCommand
+
+    def id(self) -> str:
+        return self.cid
+
+    def follow_up_action(self) -> typing.Optional[Action]:
+        pass
+
+    def reference_id(self) -> str:
+        return self.funds_pull_pre_approval.funds_pull_pre_approval_id
+
+    def validate(self, prior: typing.Optional["Command"]) -> None:
```

Comment:
The validate command is 
validate function for the pre-approvel case:
1. initial command: validate current command is valid initial status.
2. when the first command receiver sends approve / reject command back: caller gets prior command from db and calls to validate func; this function should validate current command comparing with prior is valid.

Reply:
Done.